Control/inverted_pendulum/inverted_pendulum_lqr_control.py

```diff
@@ -0,0 +1,185 @@
+"""
+Inverted Pendulum LQR control
+author: Trung Kien - letrungkien.k53.hut@gmail.com
+"""
+
+import math
+import time
+
+import matplotlib.pyplot as plt
+import numpy as np
+from numpy.linalg import inv, eig
+
+# Model parameters
+
+l_bar = 2.0  # length of bar
+M = 1.0  # [kg]
+m = 0.3  # [kg]
+g = 9.8  # [m/s^2]
+
+nx = 4  # number of state
+nu = 1  # number of input
+Q = np.diag([0.0, 1.0, 1.0, 0.0])  # state cost matrix
+R = np.diag([0.01])  # input cost matrix
+
+delta_t = 0.1  # time tick
+
+animation = True
+
+
+def main():
+    x0 = np.array([
+        [0.0],
+        [0.0],
+        [0.3],
+        [0.0]
+    ])
+
+    x = np.copy(x0)
+
+    for i in range(50):
+        # calc control input
+        u = lqr_control(x)
+
+        # simulate inverted pendulum cart
+        x = simulation(x, u)
+
+        if animation:
+            plt.clf()
+            px = float(x[0])
+            theta = float(x[2])
+            plot_cart(px, theta)
+            plt.xlim([-5.0, 2.0])
+            plt.pause(0.001)
+
+    print("Finish")
+    print(f"x={px:.2f} [m] , theta={math.degrees(theta):.2f} [deg]")
```

Comment:
If `animation` is `false`, the `px` and theta are undefined, so just `x[0]` and `x[1]` is fine?